### PR TITLE
docs: add example of dynamic layout with static page

### PR DESCRIPTION
**Pull request by tylersayshi**
you can see the time change as navigating the site which shows that the layout is indeed dynamic

The rest of the home page stays unchanged and can be cached

Example for #885

### CODE REVIEW

File: examples/25_weave_render/src/entries.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    path: '/',
+    component: HomePage,
+  }),
+
+  createPage({
+    render: 'static',
+    path: '/foo',
+    component: FooPage,
+  }),
+]);
```
Comment (dai-shi) [inline examples/25_weave_render/src/entries.tsx:28]: How about adding another route like `/nested/bar` which has static layout and dynamic page?

Comment (tylerlaws0n) [reply examples/25_weave_render/src/entries.tsx:42]: added... there is weird behavior though

When I nav to /nested/bar then click the link to go back to home or foo, I get a white screen with no content. Is this a bug? Or did I make a mistake?

Comment (dai-shi) [reply examples/25_weave_render/src/entries.tsx:42]: Does it happen both in DEV and PRD? Can be a bug.

Comment (dai-shi) [reply examples/25_weave_render/src/entries.tsx:42]: Wait, `/nested/bar` still has a dynamic root layout?

I think we should make:
- `/foo`: static layout + dynamic page
- `/nested/bar`: static layout + dynamic layout + static page

what do you think?

Comment (tylerlaws0n) [reply examples/25_weave_render/src/entries.tsx:42]: > Does it happen both in DEV and PRD? Can be a bug.

Comment (tylerlaws0n) [reply examples/25_weave_render/src/entries.tsx:42]: > Does it happen both in DEV and PRD? Can be a bug.
Yes it happens both

Comment (tylerlaws0n) [reply examples/25_weave_render/src/entries.tsx:42]: > static layout + dynamic layout + static page

How do I make two layouts on a route?

Comment (tylersayshi) [reply examples/25_weave_render/src/entries.tsx:49]: ok, I changed it. let me know if that's what you meant though. I'm not super confident with it yet.

**Comment by tylersayshi** [comment]
@dai-shi is your general experience with the _layout pattern positive? Or is it not something you use in practice very much?

File: examples/25_weave_render/src/entries.tsx (hunk 1 above)
Comment (dai-shi) [reply examples/25_weave_render/src/entries.tsx:49]: It looks different. Okay, we may not need `nested`. How about this?

```js
  createLayout({
    render: 'static',
    path: '/',
    component: HomeLayout,
  }),

  createPage({
    render: 'static',
    path: '/',
    component: HomePage,
  }),

  createLayout({
    render: 'static',
    path: '/foo',
    component: FooLayout,
  }),

  createPage({
    render: 'dynamic',
    path: '/foo',
    component: FooPage,
  }),

  createLayout({
    render: 'dynamic',
    path: '/bar',
    component: BarLayout,
  }),

  createPage({
    render: 'static',
    path: '/bar',
    component: BarPage,
  }),

  ```

**Comment by dai-shi** [comment]
> @dai-shi is your general experience with the _layout pattern positive? Or is it not something you use in practice very much?

I'm not quite sure if my usage would be similar to everyone's. With fs-router, the root layout would be necessary, but in general, I wouldn't need to use other layouts unless I care performance much.

There are a few things I would like to reconsider, and you can probably help on them:
- separation of unstable_defineRouter and createPages
  - currently, it's a bit confusing. defineRouter can be more primitive, or it's probably just naming.
- improve client cache system with defineRouter
  - Currently it's only layout/layout/page pattern, which is too related with createPages.
  - Components could be freely placed, independent from route path.
- Likewise, shouldSkip mechanism with defineRouter should be revisited.
  - It's hardly undestandable.

File: examples/25_weave_render/src/entries.tsx (hunk 1 above)
Comment (tylersayshi) [reply examples/25_weave_render/src/entries.tsx:49]: yeah, makes sense ! lgtm - changed

**Comment by tylersayshi** [comment]
> > @dai-shi is your general experience with the _layout pattern positive? Or is it not something you use in practice very much?
> 
> I'm not quite sure if my usage would be similar to everyone's. With fs-router, the root layout would be necessary, but in general, I wouldn't need to use other layouts unless I care performance much.
> 
> There are a few things I would like to reconsider, and you can probably help on them:
> 
> * separation of unstable_defineRouter and createPages
>   
>   * currently, it's a bit confusing. defineRouter can be more primitive, or it's probably just naming.
> * improve client cache system with defineRouter
>   
>   * Currently it's only layout/layout/page pattern, which is too related with createPages.
>   * Components could be freely placed, independent from route path.
> * Likewise, shouldSkip mechanism with defineRouter should be revisited.
>   
>   * It's hardly undestandable.

all sounds good to me! I don't have immediate thoughts/ideas for how to approach changing those, but I'll be thinking about what makes sense for them.

File: examples/25_weave_render/src/entries.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { lazy } from 'react';
+import { createPages } from 'waku';
+import type { PathsForPages } from 'waku/router';
+import FooPage from './components/FooPage';
+import BarPage from './components/BarPage';
+import BarLayout from './components/BarLayout';
+import FooLayout from './components/FooLayout';
+
+// The use of `lazy` is optional and you can use import statements too.
```
Comment (dai-shi) [inline examples/25_weave_render/src/entries.tsx:9]: I think we can remove this lazy usage from the examples.

Comment (tylersayshi) [reply examples/25_weave_render/src/entries.tsx:9]: I see you merged it, should I make a second PR to remove this?

Comment (dai-shi) [reply examples/25_weave_render/src/entries.tsx:9]: Yeah, please go ahead. I mean from all examples.

**Pull request merged**